bpdm-pool/src/main/kotlin/org/eclipse/tractusx/bpdm/pool/service/RequestValidationService.kt

```diff
@@ -117,51 +122,51 @@ class RequestValidationService(
         requests: Collection<LegalEntityPartnerUpdateRequest>
     ): Map<LegalEntityPartnerUpdateRequest, Collection<ErrorInfo<LegalEntityUpdateError>>> {
 
-        val legalEntityRequests = requests.map { it.legalEntity }
-        val legalAddressRequests = requests.map { it.legalAddress }
-
-        val legalEntityMetadata = metadataService.getMetadata(legalEntityRequests).toKeys()
-        val addressMetadata = metadataService.getMetadata(legalAddressRequests).toKeys()
+        val requestByLeError = validateLegalEntityUpdatesInternal(requests)
+        val errorsByRequestAddress = validateLegalEntityAddressUpdates(requests)
+        return mergeErrorMaps(requestByLeError, errorsByRequestAddress)
+    }
 
-        val legalEntityDuplicateIdentifierCandidates = getLegalEntityDuplicateIdentifierCandidates(legalEntityRequests)
-        val addressDuplicateIdentifierCandidates = getAddressDuplicateIdentifierCandidates(legalAddressRequests)
+    fun validateLegalEntityUpdatesInternal(
+        requests: Collection<LegalEntityPartnerUpdateRequest>
+    ): Map<LegalEntityPartnerUpdateRequest, Collection<ErrorInfo<LegalEntityUpdateError>>> {
 
+        val legalEntityRequests = requests.map { it.legalEntity }
+        val duplicatesValidator = ValidateLegalEntityIdentifiersDuplicated(legalEntityRequests, LegalEntityUpdateError.LegalEntityDuplicateIdentifier)
         val existingLegalEntityBpns = legalEntityRepository.findDistinctByBpnIn(requests.map { it.bpnl }).map { it.bpn }.toSet()
+        val existingBpnValidator = ValidateUpdateBpnExists(existingLegalEntityBpns, LegalEntityUpdateError.LegalEntityNotFound)
+        val legalFormValidator = ValidateLegalFormExists(legalEntityRequests,LegalEntityUpdateError.LegalFormNotFound)
+        val identifierValidator = ValidateIdentifierLeTypesExists(legalEntityRequests, LegalEntityUpdateError.LegalEntityIdentifierNotFound)
 
         return requests.flatMap { request ->
```

Comment:
Can't you use
```suggestion
        return requests.associateWith { request -> ...
           validationErrors
        }
```
without `flatMap` and `groupBy`?